includes/class-wc-payments.php

```diff
@@ -569,7 +569,9 @@ public static function init() {
 		// To avoid register the same hooks twice.
 		wcpay_get_container()->get( \WCPay\Internal\Service\DuplicatePaymentPreventionService::class )->init_hooks();
 
-		self::maybe_register_woopay_hooks();
+		// Defer registering the WooPay hooks. Later on, $wp_rewrite is used and causes a fatal error on the WooPayment Dev Tools,
+		// given that $wp_rewrite is defined right after the `plugins_loaded` action is fired. See #8857.
+		add_action( 'setup_theme', [ __CLASS__, 'maybe_register_woopay_hooks' ] );
```

Comment:
It would be good to have more eyes on this. The reasoning is explained in the commit message https://github.com/Automattic/woocommerce-payments/pull/8857/commits/69fa9ef8a2b302287dc2c315a9ec8646fdf80f33 

I've tested the full WooPay flow locally and did not experience any errors with this change.

Pinging @waclawjacek since he introduce this line initially and @malithsen since worked on something [very similar](https://github.com/Automattic/woocommerce-payments/pull/8118).

Reply:
Hi! :wave: Could you please provide some more details on how I could help here? I'm not sure what the code I introduced is, and I've been detached from this codebase for quite some time so some extra context would be much appreciated.

Reply:
Hi @waclawjacek. I believe the original line was added [in this PR of yours](https://github.com/Automattic/woocommerce-payments/pull/3481/files#diff-57ee486c4a3ba4bae478eb31255c9a968b5b4eccd04b27f512e4003440d6f821R163). I just wanted to get more eyes on this from people familiar with it. The change looks a bit sensitive since I'm delaying some hooks registrations for the reasons mentioned in https://github.com/Automattic/woocommerce-payments/commit/69fa9ef8a2b302287dc2c315a9ec8646fdf80f33. I've tested it with WooPay and double-checked that it's working. If you don't see anything to comment that's fine. I reached out to you as an extra precaution ;-)

Reply:
I see! Thanks for the explanation and the ping - I appreciate the cautiousness.

I haven't tested the code or done a super deep dive into the WP codebase but the reasoning sounds good to me. It looks like [`setup_theme` is the next hook that fires in `wp-settings.php`](https://github.com/WordPress/wordpress-develop/blob/3a9ecd2cc84fe24e2db8b537a73314c5398b5312/src/wp-settings.php#L630) after `plugins_loaded` ([which is when `WC_Payments::init()` fires right now](https://github.com/Automattic/woocommerce-payments/blob/21ba1276ee13dd160d24a2c7db5509fd9f3b200e/woocommerce-payments.php#L175)), and it looks like a good candidate to hook into, despite the hook's name being somewhat irrelevant to the logic here.

Reply:
Apologies for the delay in getting to this.

Change looks good to me. I tested WooPay locally with a WCPay account having `status=complete` and did not notice any regressions.